op-acceptor/logging/filelogger.go

```diff
@@ -197,53 +201,46 @@ func (l *FileLogger) closeAllWriters() {
 // The runID must be provided, otherwise an error is returned
 func (l *FileLogger) GetDirectoryForRunID(runID string) (string, error) {
 	if runID == "" {
-		return "", fmt.Errorf("runID is required")
+		return "", fmt.Errorf("runID cannot be empty")
 	}
-	return filepath.Join(l.logDir, fmt.Sprintf("testrun-%s", runID)), nil
+	dirName := fmt.Sprintf("testrun-%s", runID)
+	return filepath.Join(l.logDir, dirName), nil
 }
 
 // LogTestResult processes a test result through all registered sinks
 // If runID is provided, it will log to that specific run directory
 func (l *FileLogger) LogTestResult(result *types.TestResult, runID string) error {
 	if runID == "" {
-		return fmt.Errorf("runID is required")
+		return fmt.Errorf("runID cannot be empty")
 	}
 
-	// Store the result in the buffer
-	l.mu.Lock()
-	l.resultBuffer = append(l.resultBuffer, result)
-	l.mu.Unlock()
-
-	// Process the result through all sinks
+	// Feed test result to all sinks
 	for _, sink := range l.sinks {
 		if err := sink.Consume(result, runID); err != nil {
-			return err
+			return fmt.Errorf("error in sink: %w", err)
 		}
 	}
 
+	// Buffer the result for potential summary operations
+	l.resultBuffer = append(l.resultBuffer, result)
```

Comment:
Is the mutex lock/unlock no longer necessary here?

Reply:
Glad you saw/commented; this was actually dead code I should have removed.
Thanks!